weave/trace/vals.py

```diff
@@ -279,14 +286,9 @@ def __init__(
         self.server = server
         self.root = root or self
         self.parent = parent
-        self._rows: Optional[list[dict]] = None
-
-        # _prefetched_rows is a local cache of rows that can be used to
-        # avoid a remote call. Should only be used by internal code.
-        self._prefetched_rows: Optional[list[dict]] = None
 
     @property
-    def rows(self) -> list[dict]:
+    def rows(self) -> Sequence[dict]:
```

Comment:
why are we not returning a concrete type here?

Reply:
Whats the advantage of that? When users interact with the rows I think that ThreadSafeInMemorySequence should be abstracted away, we should do the heavy lifting of determining how to return the data performantly in the background.